File: flask/app/model/Cliente.py
from extensions import db

class Cliente(db.Model):
    cliente_id = db.Column(db.Integer, primary_key=True)
    nome = db.Column(db.String(100), nullable=False)
    endereco = db.Column(db.String(100), nullable=False)
    cpf = db.Column(db.String(11), nullable=False)
    rg = db.Column(db.String(11), nullable=False)
    email = db.Column(db.String(50))
    telefone_principal = db.Column(db.String(15), nullable=False)
    telefone_secundario = db.Column(db.String(15))
    observacoes = db.Column(db.String(200), nullable=False)

    processos = db.relationship('Processo', backref='cliente', lazy=True)

File: flutter/app/lib/model/cliente.dart
import 'package:controle_financeiro/model/processo.dart';

class Cliente {
  final int cliente_id;
  final String nome;
  final String endereco;
  final String cpf;
  final String rg;
  final String email;
  final String telefone_principal;
  final String telefone_secundario;
  final String observacoes;

  final List<Processo> processos;

  Cliente({
    required this.cliente_id,
    required this.nome,
    required this.endereco,
    required this.cpf,
    required this.rg,
    required this.email,
    required this.telefone_principal,
    required this.telefone_secundario,
    required this.observacoes,
    required this.processos,
  });

  factory Cliente.fromJson(Map<String, dynamic> json) {
    var list = json['processos'] as List;
    List<Processo> processosList = [];
    if (list.isNotEmpty) {
      processosList = list.map((i) => Processo.fromJson(i)).toList();
    }

    return Cliente(
      cliente_id: json['cliente_id'],
      nome: json['nome'],
      endereco: json['endereco'],
      cpf: json['cpf'],
      rg: json['rg'],
      email: json['email'],
      telefone_principal: json['telefone_principal'],
      telefone_secundario: json['telefone_secundario'],
      observacoes: json['observacoes'],
      processos: processosList,
    );
  }
}
